keps/sig-cli/2985-public-krm-functions-registry/README.md

```diff
@@ -0,0 +1,938 @@
+- A website that presents the doc and examples to the end users and allows users
+  to search and discover KRM functions.
+
+
+### Publisher
+
+A publisher can be a project/community in Kubernetes or a company. For example,
+it can be Kustomize, KubeFlow, Apple or Google.
+
+When publishing a function, the contributor can choose to publish it on behalf
+of a publisher.
+
+To publish on behalf of a k8s project/community, at least one of the maintainers
+of the project/community must approve the change.
+
+To publish on behalf of a company:
+
+* The commit must be a [verified commit](https://docs.github.com/en/github/authenticating-to-github/managing-commit-signature-verification)
+  from a person with that company’s email address.
+* If there are maintainers for this publisher, at least one of the maintainers
+  must approve the change.
+
+OWNERS files can be set up to enforce maintainer’s approvals.
+CI can be set up to enforce verifiable commit from desired email domain.
+
+### Security
+
+SIG-CLI is NOT responsible for the security of their KRM functions.
+
+Publishers are responsible for the security of their KRM functions. Publishers
+are responsible for clearly communicating the expectation (e.g. maturity) to
+their users (e.g. through a file for the publisher. We should standardize it
+eventually). For example, Kustomize wants to provide a small set of carefully
+vetted KRM functions as default and some non-default KRM functions. The former
+set can be published as _kustomize_ or _kustomize-default_ while the latter can
+be published as _kutomize-experimental_. We can also consider using hierarchy in
+publishers. E.g. _kustomize/default_ for the former set and
+_kustomize/experimental_ for the latter.
+
+If some functions don't have any publishers, the users should use it at their
+own risk.
+
+We strongly suggest users to use container as a sandboxing mechanism to run the
+KRM functions.
+
+### Trust
+
+A user should NOT trust every KRM function in the registry.
+
+Trust can be established at the publisher level. Users can choose to trust a
+publisher and use the KRM functions provided by this publisher.
+
+Publisher information can be used to aggregate KRM functions. For example, I can
+query https://krm-fn-registry.sigs.k8s.io/catalog.yaml?publisher=kustomize,kubeflow
+to find all the KRM functions offered by Kustomize and KubeFlow.
+
+### User Stories (Optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+
+As a KRM functions user, I can browse the function registry website (e.g.
+https://krm-fn-registry.sigs.k8s.io) and search the KRM function by name (e.g.
+set-labels). And I can find everything including doc, examples, homepage,
+maintainers information about the function.
```

Comment:
Suggested story from what kpt already does - I'd like to be able to discover functions form the registry via CLI.

Reply:
Added it.